### ISSUE TITLE
'Invalid instruction' for question mark in URL

### ISSUE THREAD

**Issue by loshkovskyi** [projectatomic/dockerfile_lint#46]
Hello,

In Dockerfile I have a `RUN` instruction which downloads a file:
```
RUN wget --no-check-certificate 'https://github.com/user/repo/.../binary?raw=true'
```

I'm having the following error:
```
$ dockerfile_lint -f Dockerfile --verbose

debug: Checking command : RUN
debug: Checking command : RAW=TRUE";
debug: Checking command : RUN
debug: Checking command : ENTRYPOINT
--------ERRORS---------

Line 21: -> raw=true";
ERROR: Invalid instruction.
Reference -> https://docs.docker.com/reference/builder/
```

Am I correct saying that the `?raw` is interpreted incorrectly as a instruction?
Would appreciate your advice.

**Comment by lphiri** [comment]
Hi,

If the Dockerfile works with the "docker build" command the linter shouldn't flag a syntax error.

If thats the case then its a bug in the linter.

Lindani

**Comment by loshkovskyi** [comment]
Hello Lindani,

Yes, the Dockerfile works and I'm able to build the image. It's just the linter's producing the error.

**Comment by dav1x** [comment]
@loshkovskyi Can you include the Dockerfile you are parsing? Ill take a look at it.

**Comment by loshkovskyi** [comment]
Thanks @dav1x 

Simplified Dockerfile to reproduce the problem:
```
FROM debian:jessie

RUN apt-get update -y; apt-get -y install wget

RUN wget --no-check-certificate -O k8s-template "https://github.com/davidwalter0/k8s-template/blob/release-0.1/bin/k8s-template?raw=true"
```

Issue:
```
$ dockerfile_lint -f Dockerfile 

--------ERRORS---------

Line 6: -> raw=true"
ERROR: Invalid instruction. 
Reference -> https://docs.docker.com/reference/builder/
```

**Comment by dav1x** [comment]
It looks like the regex parser is triggering a new line initiated by the ?. 

The dockerfile in question about has 5 lines the error throws on line 6. 

For example:
```
-> cat Dockerfile1
FROM debian:jessie

RUN apt-get update -y; apt-get -y install wget

RUN wget --no-check-certificate -O k8s-template 'https://github.com/davidwalter0/k8s-template/blob/release-0.1/?bin/k8s-template?raw=true'
```
Here are the results:
```
-> docker run -it --rm --privileged -v `pwd`:/root/  projectatomic/dockerfile-lint dockerfile_lint -f Dockerfile1

--------ERRORS---------

Line 6: -> bin/k8s-template
ERROR: Invalid instruction. 
Reference -> https://docs.docker.com/reference/builder/


Line 7: -> raw=true"
ERROR: Invalid instruction. 
Reference -> https://docs.docker.com/reference/builder/
```

@lphiri I believe this is a linter bug.

**Comment by dav1x** [comment]
@lphiri 

I think the problem may be in the parser.js

```
function parse(contents, options) {
    var commands = [];
    var i;
    var line;
    var lineno;
    var lines = contents.split(/[\r?\n]/);
    var parseResult;
    var partialLine = '';
```

Specifically : 
    var lines = contents.split(/[\r?\n]/);

I tested and removing the ? in the split seems to resolve the issue. Should I set up a PR for this workaround?
